include/rmm/mr/device/detail/arena.hpp

```diff
@@ -32,87 +33,116 @@
 #include <memory>
 #include <mutex>
 #include <numeric>
+#include <optional>
 #include <set>
-#include <unordered_map>
 
 namespace rmm::mr::detail::arena {
 
-/// Minimum size of a superblock (256 KiB).
-constexpr std::size_t minimum_superblock_size = 1U << 18U;
-
 /**
- * @brief Represents a chunk of memory that can be allocated and deallocated.
- *
- * A block bigger than a certain size is called a "superblock".
+ * @brief Represents a contiguous region of memory.
  */
-class block {
+class memory_span {
```

Comment:
Suggestion, rename this `byte_span` and make it work as closely as possible to `std::span<std::byte>`. Then when we adopt C++20 as minimum we can just use the standard types.